### PR TITLE
Written guide for elasticsearch

### PR DESCRIPTION
**Pull request by rajujha373**
with reference to #3803

### CODE REVIEW

File: docs/guides/run-elasticsearch.rst [hunk 1]
```diff
@@ -0,0 +1,84 @@
+Enabling Elasticsearch on the local server
```
Comment (stsewd) [inline docs/guides/run-elasticsearch.rst]: Use `Title Case` for the title

File: docs/guides/run-elasticsearch.rst [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+==========================================
+
+Read the Docs has been using Elasticsearch which is a platform for distributed search and analysis of data in real time. To enable the search feature on your local installation, you need to install the elasticsearch locally and run the Elastic server. 
+
+Installation has been mainly divided into following steps.
+
+1. Installing Java
```
Comment (stsewd) [inline docs/guides/run-elasticsearch.rst]: Isn't better to link to official resources for this step? So we don't  need to maintain this at the future or write it for every OS.

File: docs/guides/run-elasticsearch.rst [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	(READTHEDOCS)$ java -version
+
+The result should be something like this::
+
+	openjdk version "1.8.0_151"
+	OpenJDK Runtime Environment (build 1.8.0_151-8u151-b12-0ubuntu0.16.04.2-b12)
+	OpenJDK 64-Bit Server VM (build 25.151-b12, mixed mode)
+
+
+2. Downloading and installing Elasticsearch
```
Comment (stsewd) [inline docs/guides/run-elasticsearch.rst]: Same here, linking to the official installation guide could be better

File: docs/guides/run-elasticsearch.rst (hunk 2 above)
Comment (rajujha373) [reply docs/guides/run-elasticsearch.rst]: ok @stsewd , I will do the required changes

**Comment by xrmx** [comment]
@rajujha373 Have you tried the elasticsearch ppa? Am not sure you need to add a ppa for openjdk 8 on a currently supported distribution.

**Comment by rajujha373** [comment]
@xrmx I have not tried elasticsearch ppa, I just downloaded the package and ran the server manually. I guess I am going to give it a try.

File: docs/guides/run-elasticsearch.rst [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+-------------------------------------------
+
+Elasticsearch can be downloaded directly from elastic.co in zip, tar.gz, deb, or rpm packages. For Ubuntu, it's best to use the deb (Debian) package which will install everything you need to run Elasticsearch.
+
+RTD currently uses elasticsearch 1.x which can be easily downloaded and installed from the official website (http://elastic.co).
+
+Download it in a directory parallel to where the readthedocs.org project has been stored.::
```
Comment (stsewd) [inline docs/guides/run-elasticsearch.rst]: This is strictly necessary? What about the people that install it with a package manager? (also I think this step is visible on the elastic search official guide)

File: docs/guides/run-elasticsearch.rst [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+Your directory structure after the above command would be:
+
+READTHEDOCS
+   |- readthedocs.org
+   |- elasticsearch-1.3.8
+
+3. Running Elasticsearch from command line
+------------------------------------------
+
+Goto elasticsearch home directory.::
```
Comment (stsewd) [inline docs/guides/run-elasticsearch.rst]: Same here, maybe just saying that executing the server with `elasticsearch` is better here.

File: docs/guides/run-elasticsearch.rst (hunk 4 above)
Comment (rajujha373) [reply docs/guides/run-elasticsearch.rst]: After some research, I found out that there are various ways to start the elastic server and the steps I have written works as well but requires a bit more work than the installation with package manager. 
I guess i'm gonna re-write the guide for installing and setting up elastic server via package manager.  :)

**Comment by rajujha373** [comment]
Hey @stsewd I have re-written the guide for installation of elasticsearch in local system. The code has been a mess currently but once you finalise the changes, I can create a new PR if you wish. 
Do notify me if any changes are necessary :)

**Comment by stsewd** [comment]
@rajujha373 not sure why the diff shows unrelated files, also always create a new branch when making changes (I see you did it on your master branch).

**Comment by rajujha373** [comment]
Yeah I too noticed it after making the changes. I'll keep this in mind next time :) @stsewd

**Pull request closed without merging**